### PR TITLE
Remove some redundances from Microsoft.CSharp

### PR DESCRIPTION
**Pull request by JonHanna**
* Remove dead path from `CheckAccessCore`.

If `aggWhere` was null we return, otherwise last for must happen at least once. Within that if `atsThru` is null we return, therefore later path for `atsThru` being null cannot be hit.

* Remove dead test from `TryReportLvalueFailure`

If `!isNested` we are on the first loop so if `pObject != null` and `!(expr is ExprField)` then we know statically that `expr is ExprProperty` as only paths for it being `ExprField` or `ExprProperty` set `pObject`, so remove redundant test.

* Remove parameter from `bindFloatOp` that is always overwritten.

* Remove parameter from `LScalar` that is always overwritten.

* Reduce duplication in `BindIncOpCore`.

* Remove some unused assignments.

### CODE REVIEW

**Comment by JonHanna** [comment]
Closing because will clash with some slightly bigger fish I'd like to fry first.

**Comment by JonHanna** [comment]
Test OSX x64 Debug Build

**Comment by OmarTawfik** [comment]
>         private void TryReportLvalueFailure(Expr expr, CheckLvalueKind kind)

nit: Is this comment still valid?

---
Refers to: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Binding/ErrorReporting.cs:60 in 1cd979b. [](commit_id = 1cd979bc74ac9a56bbbafde3615963c7d524d27b, deletion_comment = False)

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs [hunk 1]
```diff
@@ -504,15 +502,10 @@ private ExprCall BindToMethod(MethWithInst mwi, Expr pArguments, ExprMemberGroup
...
-            CType pReturnType = null;
-            if ((flags & (MemLookFlags.Ctor | MemLookFlags.NewObj)) == (MemLookFlags.Ctor | MemLookFlags.NewObj))
-            {
-                pReturnType = mwi.Ats;
-            }
-            else
-            {
-                pReturnType = GetTypes().SubstType(mwi.Meth().RetType, mwi.GetType(), mwi.TypeArgs);
-            }
+            CType pReturnType =
```
Comment (OmarTawfik) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs]: I'd suggest keeping the old if statement (with removing `= null`) for readability.

**Comment by JonHanna** [comment]
Which comment are you referring to above, @OmarTawfik ?

**Comment by OmarTawfik** [comment]
@JonHanna the following on `TryReportLvalueFailure`
```csharp
// Return true if we actually report a failure.
```
The only caller seems to return the same Boolean value since this has a void return type.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs [hunk 2]
```diff
@@ -565,15 +563,11 @@ private ExprBinOp BindLiftedStandardBinOp(BinOpArgInfo info, BinOpFullSig bofs,
             // method so that we error report (ie divide by zero etc), and then we store in the resulting
             // binop that we have a lifted operator.
 
-            Expr pArgument1 = null;
-            Expr pArgument2 = null;
-            Expr nonLiftedArg1 = null;
-            Expr nonLiftedArg2 = null;
             Expr nonLiftedResult = null;
-            CType resultType = null;
+            CType resultType;
```
Comment (OmarTawfik) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs]: >resultType [](start = 18, length = 10)

Move next to usage?

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs [hunk 3]
```diff
@@ -1669,20 +1659,20 @@ private ExprOperator BindIntUnaOp(ExpressionKind ek, EXPRFLAG flags, Expr arg)
         /*
             Handles standard binary floating point (float, double) based operators.
         */
-        private ExprOperator BindRealBinOp(ExpressionKind ek, EXPRFLAG flags, Expr arg1, Expr arg2)
+        private ExprOperator BindRealBinOp(ExpressionKind ek, EXPRFLAG _, Expr arg1, Expr arg2)
```
Comment (OmarTawfik) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs]: >EXPRFLAG _, [](start = 62, length = 11)

Remove the parameter?

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs [hunk 4]
```diff
@@ -1669,20 +1659,20 @@ private ExprOperator BindIntUnaOp(ExpressionKind ek, EXPRFLAG flags, Expr arg)
...
-            return bindFloatOp(ek, flags, arg1, arg2);
+            return bindFloatOp(ek, arg1, arg2);
         }
 
 
         /*
             Handles standard unary floating point (float, double) based operators.
         */
-        private ExprOperator BindRealUnaOp(ExpressionKind ek, EXPRFLAG flags, Expr arg)
+        private ExprOperator BindRealUnaOp(ExpressionKind ek, EXPRFLAG _, Expr arg)
```
Comment (OmarTawfik) [inline src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs]: > EXPRFLAG _ [](start = 61, length = 11)

Remove the parameter?

Comment (JonHanna) [reply src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs]: It's included in an array of delegates, so it has to match the signature.

**Comment by JonHanna** [comment]
`TryReportLvalueFailure` was made void in a previous change, so the comment should really have been removed with that, but no harm including it in this.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs (hunk 1 above)
Comment (JonHanna) [reply src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs]: I think shortening the test part of the conditional makes it more readable, but readability is subjective and I've a high tolerance for conditionals over if/else (blame coding javascript in the days when most customers where on dialup, minifiers weren't common, and managers set small kilobyte limits) so if you'd still prefer the if/else I'll go with that.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs (hunk 3 above)
Comment (JonHanna) [reply src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs]: Another that has to match a delegate signature.

File: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs (hunk 1 above)
Comment (JonHanna) [reply src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs]: Scratch that. That's not equivalent at all. Turning back into an if/else.

**Pull request merged**